src/python/pants/backend/go/go_sources/analyze_package/string_utils.go

```diff
@@ -0,0 +1,89 @@
+/* Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+ * Licensed under the Apache License, Version 2.0 (see LICENSE).
+ */
+
+/** Vendored from Go
```

Comment:
Maybe a link to where it comes from?

Reply:
Note that I only did a cursory review of this file, assuming it's well vetted from Go.

Reply:
The relevant functions were copied verbatim (although I had to rename `Cut` to `stringCut` since it was only added recently to the Go standard library and is not available on 1.17 and earlier).